handlers/hierarchy.go

```diff
@@ -74,55 +75,57 @@ func (f *Filter) HierarchyUpdate(w http.ResponseWriter, req *http.Request) {
 		return
 	}
 
-	var h hierarchy.Model
+	var wg sync.WaitGroup
+	wg.Add(1)
 
-	if len(code) > 0 {
-		h, err = f.HierarchyClient.GetChild(ctx, fil.InstanceID, name, code)
-	} else {
-		if name == "geography" {
-			h, err = f.flattenGeographyTopLevel(ctx, fil.InstanceID)
-		} else {
-			h, err = f.HierarchyClient.GetRoot(ctx, fil.InstanceID, name)
-		}
+	go func() {
+		defer wg.Done()
 
-		// We include the value on the root as a selectable item, so append
-		// the value on the root to the child to see if it has been removed by
-		// the user
-		h.Children = append(h.Children, hierarchy.Child{
-			Links: h.Links,
-		})
-	}
-	if err != nil {
-		log.Event(ctx, "failed to get hierarchy node", log.Error(err), log.Data{"filter_id": filterID, "dimension": name, "code": code})
-		setStatusCode(req, w, err)
-		return
-	}
+		var h hierarchy.Model
+		var err error
+		if len(code) > 0 {
+			h, err = f.HierarchyClient.GetChild(ctx, fil.InstanceID, name, code)
+		} else {
+			if name == "geography" {
+				h, err = f.flattenGeographyTopLevel(ctx, fil.InstanceID)
+			} else {
+				h, err = f.HierarchyClient.GetRoot(ctx, fil.InstanceID, name)
+			}
 
-	// unique options to be posted. Using a map for faster access and uniqueness
-	mapUniqueOptions := make(map[string]struct{})
+			// We include the value on the root as a selectable item, so append
+			// the value on the root to the child to see if it has been removed by
+			// the user
+			h.Children = append(h.Children, hierarchy.Child{
+				Links: h.Links,
+			})
+		}
+		if err != nil {
+			log.Event(ctx, "failed to get hierarchy node", log.Error(err), log.Data{"filter_id": filterID, "dimension": name, "code": code})
+			setStatusCode(req, w, err)
+			return
+		}
 
-	// Get existing dimension options from filterAPI
-	existingOptions, err := f.FilterClient.GetDimensionOptions(req.Context(), userAccessToken, "", collectionID, filterID, name)
-	if err != nil {
-		log.Event(ctx, "failed to get dimension options", log.Error(err))
-		setStatusCode(req, w, err)
-		return
-	}
+		// TODO when PATCH endpoint is implemented in Filter API, we can send an array of delete operations for all options found in child items but not in the request form
```

Comment:
I think we might still have out of memory problems if we call GetDimensionOptions. I would suggest that once we have the `PATCH` endpoint in Filter API, we send the remove operations for options found in children and not found in the request form, regardless of their existence in Filter API.

Reply:
Surely we know what became unchecked to know what needs removing?

And yes there should not be a need to do a GetDimensionOptions in this handler